prov/gni/include/gnix.h

```diff
@@ -494,15 +506,76 @@ struct gnix_fid_ep {
 				   (type == FI_EP_MSG))
 
 /**
+ * gnix_fid_sep struct
+ *
+ * @var ep_fid          embedded struct fid_ep field
+ * @var domain          pointer to domain used to create the sep instance
+ * @var info            pointer to dup of info struct supplied to fi_scalable_ep
+ *                      operation
+ * @var op_flags        quick access for op_flags for tx/rx contexts
+ *                      instantiated using this sep
+ * @var caps            quick access for caps for tx/rx contexts instantiated
+ *                      using this sep
+ * @var cdm_id_base     base cdm id to use for tx/rx contexts instantiated
+ *                      using this sep
+ * @var ep_table        array of pointers to EPs used by the rx/tx contexts
+ *                      instantiated using this sep
+ * @var tx_ep_table     array of pointers to tx contexts instantiated using
+ *                      this sep
+ * @var rx_ep_table     array of pointers to rx contexts instantiated using
+ *                      this sep
+ * @var cm_nic          gnix cm nic associated with this SEP.
+ * @var av              address vector bound to this SEP
+ * @var my_name         ep name for this endpoint
+ * @var sep_lock        lock protecting this sep object
+ * @var ref_cnt         ref cnt on this object
+ */
+struct gnix_fid_sep {
+	struct fid_ep ep_fid;
+	enum fi_ep_type type;
+	struct fid_domain *domain;
+	struct fi_info *info;
+	uint64_t op_flags;
+	uint64_t caps;
+	uint32_t cdm_id_base;
+	struct fid_ep **ep_table;
+	struct fid_ep **tx_ep_table;
+	struct fid_ep **rx_ep_table;
+	struct gnix_cm_nic *cm_nic;
+	struct gnix_fid_av *av;
+	struct gnix_ep_name my_name;
+	fastlock_t sep_lock;
+	struct gnix_reference ref_cnt;
+};
+
+/**
+ * gnix_fid_trx struct
+ *
+ * @var ep_fid          embedded struct fid_ep field
+ * @var ep              pointer to gnix_fid_ep used by this tx/rx context
+ * @var op_flags        op flags for this tx context
+ * @var caps            caps for this tx context
+ * @var ref_cnt         ref cnt on this object
+ */
+struct gnix_fid_trx {
+	struct fid_ep ep_fid;
+	struct gnix_fid_ep *ep;
+	struct gnix_fid_sep *sep;
+	uint64_t op_flags;
+	uint64_t caps;
+	struct gnix_reference ref_cnt;
+};
+
+/**
  * gnix_fid_stx struct
  * @note - this is essentially a dummy structure for the GNI
  *         provider.  Added as a convenience for those consumers
  *         who need to use the fid_stx construct for other
  *         providers.
  *
- * @var stx_fid              embedded struct fid_stx field
- * @var domain               pointer to domain used to create the stx instance
- * @var ref_cnt              ref cnt on this object
+ * @var stx_fid         embedded struct fid_stx field
+ * @var domain          pointer to domain used to create the stx instance
+ * @var ref_cnt         ref cnt on this object
```

Comment:
I don't see ref_cnt in the structure, plus some variables such as capacity aren't documented.